### PR TITLE
fix(auth): correct placeholder layout and styling in SSInput

### LINKED ISSUE #1994: UI: Fix input field overflow on Login page
### Is there an existing issue for this?

- [x] I have searched the existing issues

### What would you like to improve?

The login form inputs overflow beyond the boundaries of the authentication card, resulting in an inconsistent and visually broken layout.

Current behavior

Email and password input fields extend outside the login container.
Input widths do not properly align with the parent card.
The form layout appears visually inconsistent.

Expected behavior

Input fields should remain fully contained within the login card.
Input widths should align correctly with the parent container.
The login form should maintain a clean and consistent layout across supported screen sizes.

### Why does this matter?

This issue affects the visual appearance and usability of the login page. Users may perceive the form as broken or unpolished when input fields extend beyond their intended container.

Fixing this would improve UI consistency and provide a better user experience across devices and screen sizes.

### Screenshots or screen recording (if applicable)

Image

### Record

- [x] I agree to follow this project's Code of Conduct
- [x] I want to work on this issue

I'd be happy to work on this issue under GSSoC'26 👍

### PR DESCRIPTION
**Pull request by krishpatel2-prog**
## Screenshot 

**Before:** *Screenshot 2026-06-02 230702*

**After:** *Screenshot 2026-06-02 231552*

---

## What this PR does

This PR fixes the layout issue on the Login page where input fields extend beyond the boundaries of the authentication card.

### Changes made

* Corrected input field sizing to properly respect the parent container width.
* Fixed layout overflow affecting the login form.
* Improved visual consistency between form elements and the authentication card.
* Preserved existing login functionality and behavior.

The change is purely related to UI/layout and does not affect authentication logic, validation, or API interactions.

---

## Type of change

* [x] Bug fix
* [ ] New feature
* [ ] Code refactor
* [ ] Documentation update

---

## Related issue

Closes #1994 

---

## Checklist

* [x] I have added screenshots or a recording when they help explain this PR.
* [x] I have filled in the related issue number when there is one.
* [x] I have tested my changes.
* [x] I have done a self-review of the code.

### CODE REVIEW

**Comment by krishpatel2-prog** [comment]
@ronisarkarexe 

Just a quick note: this PR only contains a small frontend UI fix related to the login form layout.

The failing lint/typecheck errors appear to be coming from files outside the scope of this PR. No backend, configuration, dependency, or CI-related files were modified.

Happy to make any required changes, but the current CI failures seem unrelated to the changes introduced here. Thanks!

**Comment by ronisarkarexe** [comment]
Please give a star to support my project.

**Comment by krishpatel2-prog** [comment]
## 🤖 AI Code Review by ReviewPilot
### 📌 Summary
This pull request fixes a layout issue in the SSInput component, correcting the sizing of input fields to respect the parent container width and improving visual consistency. The changes are purely related to UI/layout and do not affect authentication logic, validation, or API interactions. The fix is applied by adjusting the CSS classes and styling of the input fields and surrounding elements.

### ⚠️ Issues
- The input field's type is not properly handled when switching between password and text types
- The error message is not properly displayed when an error occurs

### 💡 Suggestions
- Consider adding more robust testing for edge cases, such as extremely long input values or unusual character combinations
- Use a more consistent naming convention for CSS classes to improve maintainability

### 🔥 Risk Level: `LOW`
> The changes consist of 40 additions and 29 deletions in a single file, with no modifications to core logic or authentication functionality. The changes are primarily related to UI/layout, which is a relatively isolated and low-risk area.

---
*Powered by ReviewPilot 🤖*

File: frontend/src/components/ui-component/ss-input/ss-input.tsx [hunk 1]
```diff
@@ -34,52 +34,63 @@ const SSInput = <T extends FieldValues>({
...
-            error
-              ? "border-2 border-red-500"
-              : "border-gray-300 dark:border-slate-700 focus:border-indigo-500 focus:ring-2 focus:ring-indigo-500/20 focus:outline-none"
-          }`}
-          placeholder={placeholder}
-          autoComplete={autoComplete}
-          autoFocus={autoFocus}
-          {...register(name, validation)}
-        />
+       <input
```
Comment (krishpatel2-prog) [inline frontend/src/components/ui-component/ss-input/ss-input.tsx:58]: 🤖 The input field is missing the 'required' attribute, consider adding it if the field is required.

File: frontend/src/components/ui-component/ss-input/ss-input.tsx [hunk 2]
```diff
@@ -34,52 +34,63 @@ const SSInput = <T extends FieldValues>({
...
+    
+    aria-label={showPassword ? "Hide password" : "Show password"}
+    title={showPassword ? "Hide password" : "Show password"}
+
+  >
+    <i className={showPassword ? "fi fi-rr-eye" : "fi fi-rr-eye-crossed"}></i>
+  </button>
+)}
       </div>
       {error && (
```
Comment (krishpatel2-prog) [inline frontend/src/components/ui-component/ss-input/ss-input.tsx:87]: 🤖 The error message is not being properly truncated, consider adding an ellipsis if the message is too long.
